src/android/de/martinreinhardt/cordova/plugins/hotspot/HotSpotPlugin.java

```diff
@@ -990,6 +1000,25 @@ private boolean connectToWifiNetwork(final CallbackContext callback, final Strin
     return true;
   }
 
+  /**
+   * Determines if connection is a captive portal and returns true to app if it is.
+   * Otherwise, returns false.
+   */
+  private void isCaptivePortalConnection(final CallbackContext callback) {
+    try {
+      final Activity activity = this.cordova.getActivity();
+      WifiHotSpots hotspot = new WifiHotSpots(activity);
+      boolean isCaptivePortal = hotspot.isCaptivePortalConnection();
+      JSONObject result = new JSONObject();
+      result.put("isCaptivePortal", isCaptivePortal);
+      Log.d("RAVEN", "Is captive portal - " + isCaptivePortal);
+      callback.success(result);
+    } catch (JSONException e) {
+      Log.d("RAVEN", "ERROR - failed to determine if network had a captive portal. Assuming that it doesn't.")
+      callback.success(false);
```

Comment:
this can be `error` callback with the error message or the msg that is printed above.

Reply:
I was thinking that we wouldn't want to throw an error here either. If for some reason this was continually throwing a JSONException when trying to pass pack whether or not it is a captive portal, we wouldn't want to disconnect their internet connection over this. It would be better to silently eat the error